src/app/core/params.ts

```diff
@@ -4,9 +4,9 @@ import Parameter from "./parameter";
  * Params class
  */
 export default class Params {
-    private unique: Array<string>;
-    private heritable: Array<string>;
-    private parameters: {[index: string]: Parameter};
+    private __unique: Map<string,string>;
+    private __heritable: Map<string,string>;
```

Comment:
What is stored in the index here ?

Reply:
The index is the name of the property, the value is the type

Reply:
Okay it's not obvious why this should be like this since the type is part of the given parameter object.